packages/frontend/src/state/lp/hooks.ts

```diff
@@ -13,6 +13,62 @@ import { ethers } from 'ethers'
 import { useCallback } from 'react'
 import { useGetDebtAmount, useGetTwapSqueethPrice, useGetVault } from '../controller/hooks'
 
+/*** CONSTANTS ***/
+const TICK_SPACE = 60
+const COLLAT_RATIO = 2
+
+/*** ACTIONS ***/
+
+// Opening a mint and LP position and depositing
+export const useOpenPositionDeposit = () => {
+  const { squeethPool } = useAtomValue(addressesAtom)
+  const address = useAtomValue(addressAtom)
+  const contract = useAtomValue(controllerHelperHelperContractAtom)
+  const handleTransaction = useHandleTransaction()
+  const getTwapSqueethPrice = useGetTwapSqueethPrice()
+  const getDebtAmount = useGetDebtAmount()
+  const openPositionDeposit = useAppCallback(
+    async (squeethToMint: BigNumber, lowerTickInput: number, upperTickInput: number, vaultId: number, onTxConfirmed?: () => void) => {
+      if (!contract || !address) return null
+      
+      const mintWSqueethAmount = fromTokenAmount(squeethToMint, OSQUEETH_DECIMALS)
+      const squeethPricePromise = getTwapSqueethPrice()
+      const ethDebtPromise = getDebtAmount(mintWSqueethAmount)
+      const [squeethPrice, ethDebt] = await Promise.all([squeethPricePromise, ethDebtPromise])
+
+      const collateralToMint = ethDebt.multipliedBy(COLLAT_RATIO)
+      const collateralToLp = mintWSqueethAmount.multipliedBy(squeethPrice)
+
```

Comment:
Have example fns for amounts getAmounts, getNewAmounts at end here:  https://colab.research.google.com/drive/1oslr7uAN27q4-51iwY2X5lPqEDGTew32#scrollTo=1RF_rzrswNQL

but I think we got these in later PRs anyway